### PR TITLE
Update documentation according to the new powsybl-hpc repository

### PR DESCRIPTION
**Pull request by mathbagu**

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -12,9 +12,12 @@ open source innovation projects within the energy and electricity sectors.
 
 # Open source
 Powsybl in an open source framework licensed under the [Mozilla Public License 2.0](https://www.mozilla.org/en-US/MPL/2.0/).
-The source code is hosted on GitHub in two different repositories:
+The source code is hosted on GitHub in several repositories:
 - [powsybl-core](https://github.com/powsybl/powsybl-core) contains all the core features
+- [powsybl-math-native](https://github.com/powsybl/powsybl-math-native) are natives modules (JNI) used by powsybl-math modules (LU factorization)
```
Comment (sylvlecl) [inline index.md]: May in more details? "are native module (C++ wrapped with JNI compiled for multiple platforms and and delivered in jars)"

File: index.md [hunk 2]
```diff
@@ -12,9 +12,12 @@ open source innovation projects within the energy and electricity sectors.
...
+- [powsybl-hpc](https://github.com/powsybl/powsybl-hpc) contains all modules relatives to high performance computing
```
Comment (sylvlecl) [inline index.md]: Maybe add : "in particular computations distribution on a cluster of machines"

File: index.md [hunk 3]
```diff
@@ -12,9 +12,12 @@ open source innovation projects within the energy and electricity sectors.
...
 - [powsybl-gse](https://github.com/powsybl/powsybl-gse) contains the JavaFX UI
+- [powsybl-incubator](https://github.com/powsybl/powsybl-incubator) contains incubating modules that will be move once mature
```
Comment (sylvlecl) [inline index.md]: type "move**d**"

File: docs/configuration/itools.md [hunk 4]
```diff
@@ -47,6 +39,12 @@ The `itools_config_dir` property is deprecated since V2.2.0. Use the `powsybl_co
 ## itools_config_name
 The `itools_config_name` property is deprecated since V2.2.0. Use the `powsybl_config_name` property instead. 
 
+## mpi_hosts
```
Comment (sylvlecl) [inline docs/configuration/itools.md]: Maybe add "deprecated" in the title to make it even more clear ?

Comment (mathbagu) [reply docs/configuration/itools.md]: The title of this section is _Deprecated properties_. It's strange to repeat it everywhere. I use this convention also in configuration modules.

Comment (sylvlecl) [reply docs/configuration/itools.md]: You're right, I have missed it in the diff.

**Pull request merged**